Zend/zend_compile.c

```diff
@@ -6956,8 +6956,16 @@ static zend_bool zend_try_ct_eval_array(zval *result, zend_ast *ast) /* {{{ */
 					zend_symtable_update(Z_ARRVAL_P(result), Z_STR_P(key), value);
 					break;
 				case IS_DOUBLE:
-					zend_hash_index_update(Z_ARRVAL_P(result),
-						zend_dval_to_lval(Z_DVAL_P(key)), value);
+					if (ZEND_DOUBLE_FITS_LONG(Z_DVAL_P(key))) {
+						zend_hash_index_update(Z_ARRVAL_P(result),
+							zend_dval_to_lval(Z_DVAL_P(key)), value);
+					} else {
+						zval tmp;
+						ZVAL_COPY(&tmp, key);
+						convert_to_string(&tmp);
+						zend_symtable_update(Z_ARRVAL_P(result), Z_STR(tmp), value);
+						zval_dtor(&tmp);
```

Comment:
I think you could shorten this to one line by using `zval_get_string`:

    zend_symtable_update(Z_ARRVAL_P(result), zval_get_string(key), value);

Likewise the other instances.

Reply:
Using zval_get_string is preferable, but you'd still have to release the string afterward. The hash API does not release the string.

Reply:
…yep, silly me.